index.tsx

```diff
@@ -0,0 +1,346 @@
+  remoteVideoUrl?: string;
+  video?: string;
+};
+
+type ActivityItemSource = {
+  placeholderItem: ActivityItem;
+  item: { markupAsPDF: ActivityItem } | { [key: string]: ActivityItem };
+  subject?: Partial<IActivityType> | { [key: string]: string };
+  dataTypeIdentifier?: Partial<IActivityType> | { [key: string]: string };
+  thumbnailImage?: Partial<IActivityType> | { [key: string]: string };
+  linkMetadata?: LinkMetadata;
+};
+
+type OpenReturn = { app?: string; dismissedAction?: boolean };
+type AppMessageReturn = { app: ActivityType; message: ActivityType };
+type ShareSingleReturn = { message: string; isInstalled?: boolean };
+
+const requireAndAskPermissions = async (options: Options | MultipleOptions) => {
+  if ((options.url || options.urls) && Platform.OS === 'android') {
+    const urls: Array<string> = options.urls || (options.url ? [options.url] : []);
+    try {
+      const resultArr = await Promise.all(
+        urls.map(
+          (url) =>
+            new Promise((res, rej) => {
+              NativeModules.RNShare.isBase64File(
+                url,
+                (e: unknown) => {
+                  rej(e);
+                },
+                (isBase64: boolean) => {
+                  res(isBase64);
+                },
+              );
+            }),
+        ),
+      );
+
+      const requirePermission = resultArr.includes(true);
+      if (!requirePermission) {
+        return Promise.resolve(true);
+      }
+      const hasPermission = await PermissionsAndroid.check(
+        PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+      );
+      if (hasPermission) {
+        return Promise.resolve(true);
+      }
+      const result = await PermissionsAndroid.request(
+        PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+      );
+      if (result === PermissionsAndroid.RESULTS.GRANTED) {
+        return Promise.resolve(true);
+      }
+      throw new Error('Write Permission not available');
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
+  return Promise.resolve(true);
+};
+
+class RNShare {
+  static Button: any;
+  static ShareSheet: any;
+  static Overlay: any;
+  static Sheet: any;
+  static Social = {
+    FACEBOOK: NativeModules.RNShare.FACEBOOK || 'facebook',
+    FACEBOOK_STORIES: NativeModules.RNShare.FACEBOOK_STORIES || 'facebook-stories',
+    PAGESMANAGER: NativeModules.RNShare.PAGESMANAGER || 'pagesmanager',
+    TWITTER: NativeModules.RNShare.TWITTER || 'twitter',
+    WHATSAPP: NativeModules.RNShare.WHATSAPP || 'whatsapp',
+    INSTAGRAM: NativeModules.RNShare.INSTAGRAM || 'instagram',
+    INSTAGRAM_STORIES: NativeModules.RNShare.INSTAGRAM_STORIES || 'instagramstories',
+    GOOGLEPLUS: NativeModules.RNShare.GOOGLEPLUS || 'googleplus',
+    EMAIL: NativeModules.RNShare.EMAIL || 'email',
+    PINTEREST: NativeModules.RNShare.PINTEREST || 'pinterest',
+    LINKEDIN: NativeModules.RNShare.LINKEDIN || 'linkedin',
+    SMS: NativeModules.RNShare.SMS || 'sms',
+  };
```

Comment:
What do you think about a `enum` to reference these values instead of hardcoded strings?